src/proposal.ts

```diff
@@ -340,16 +345,18 @@ export class Proposal implements IStateful<IProposalState> {
     )
   }
 
+  public stakingToken() {
+    // TODO: the token to stake with from the proposal, instead ot he generic one
+    return new Token(this.context.getContract('GEN').options.address, this.context)
+  }
+
   public stakes(options: IStakeQueryOptions = {}): Observable<IStake[]> {
     options.proposal = this.id
     return Stake.search(this.context, options)
   }
 
   public stake(outcome: ProposalOutcome, amount: number ): Operation<Stake> {
-    // TODO: cf. vote() function: get the contract from the proposal
-    const votingMachine = this.context.getContract('GenesisProtocol')
-
-    const stakeMethod = votingMachine.methods.stake(
+    const stakeMethod = this.votingMachine().methods.stake(
       this.id,  // proposalId
       outcome, // a value between 0 to and the proposal number of choices.
       amount // amount the amount to stake with . if _amount == 0 it will use all staker reputation.
```

Comment:
reputation has nothing to do with staking, and you always have to choose an amount to stake

Reply:
true